extensions/interactions/AlgebraicExpressionInput/directives/oppia-interactive-algebraic-expression-input.component.spec.ts

```diff
@@ -18,43 +18,25 @@
  */
 
 import { DeviceInfoService } from 'services/contextual/device-info.service.ts';
-import { GuppyConfigurationService } from
-  'services/guppy-configuration.service.ts';
-import { GuppyInitializationService } from
-  'services/guppy-initialization.service.ts';
-import { MathInteractionsService } from 'services/math-interactions.service.ts';
-import { WindowRef } from 'services/contextual/window-ref.service.ts';
+import { AlgebraicExpressionInputInteractionComponent } from './oppia-interactive-algebraic-expression-input.component';
+import { ComponentFixture, waitForAsync, TestBed } from '@angular/core/testing';
+import { CurrentInteractionService } from 'pages/exploration-player-page/services/current-interaction.service';
+import { GuppyInitializationService } from 'services/guppy-initialization.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
 
-require(
-  'interactions/AlgebraicExpressionInput/directives/' +
-  'algebraic-expression-input-rules.service.ts');
-require(
-  'pages/exploration-player-page/services/current-interaction.service.ts');
-require(
-  'interactions/AlgebraicExpressionInput/directives/' +
-  'oppia-interactive-algebraic-expression-input.component.ts');
-
-describe('AlgebraicExpressionInputInteractive', function() {
-  let ctrl = null, $window = null;
-  let mockCurrentInteractionService = {
-    onSubmit: function(answer, rulesService) {},
-    registerCurrentInteraction: function(submitAnswerFn, validateExpressionFn) {
-      submitAnswerFn();
-    }
-  };
-  let mockAlgebraicExpressionInputRulesService = {};
+describe('AlgebraicExpressionInputInteractive', () => {
+  let component: AlgebraicExpressionInputInteractionComponent;
+  let fixture: ComponentFixture<AlgebraicExpressionInputInteractionComponent>;
+  let windowRef: WindowRef;
+  let guppyInitializationService = null;
+  let deviceInfoService = null;
```

Comment:
You can remove the `null`s here and maybe can add types.
```suggestion
  let guppyInitializationService;
  let deviceInfoService;
```

Reply:
Done!